pkg/signals/signals.go

```diff
@@ -28,7 +30,9 @@ func Context() context.Context {
 
 			select {
 			case <-signalCtx.Done():
+				logrus.Info("received the done signal")
```

Comment:
Hmm, these look like debug artifacts that were added in the initial, merged PR that made it's way through code review. I'm not entirely sure I see the value in logging the current state of signal handling, at least at the info-level log level, but not the end of the world. Happy to modify that cherrypick(-ed) commit and remove this additional logging if anyone has a strong opinion.